src/sdltiles.cpp

```diff
@@ -2963,8 +2978,22 @@ void catacurses::init_interface()
 
     find_videodisplays();
 
-    TERMINAL_WIDTH = get_option<int>( "TERMINAL_X" );
-    TERMINAL_HEIGHT = get_option<int>( "TERMINAL_Y" );
+    std::string scaling_factor_setting = get_option<std::string>( "SCALING_FACTOR" );
```

Comment:
Why can't you get option value as an `int` and set `scaling_factor` based directly on that value, something like
```
scaling_factor = get_option<int>( "SCALING_FACTOR" );
```

Reply:
I thought so too, but that causes an abnormal termination for some reason.